packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -108,11 +112,21 @@ abstract class Route<T> {
 
   /// The route should remove its overlays and free any other resources.
   ///
-  /// A call to didPop() implies that the Route should call dispose() itself,
-  /// but it is possible for dispose() to be called directly (e.g. if the route
-  /// is replaced, or if the navigator itself is disposed).
+  /// This route is no longer referenced by the navigator.
```

Comment:
It's already `@protected`, so it's really only callable by navigator or a subclass.  None of the other lifecycle callbacks have that comment...